community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -649,11 +649,14 @@ public long scanForHighId() throws IOException
             long nextPageId = storeFile.getLastPageId();
```

Comment:
No, `rebuildIdGeneratorSlow` does not suffer from this because 1) it's using a shared write lock, and 2) it resets its state and rescans the whole page within its `shouldRetry` loop.